### PR TITLE
prov/efa: Atomic VLA -> Malloc

### PR DESCRIPTION
**Pull request by a-szegel**
Atomics where implemented with VLA (variable length arrays) which are part of the C standard, but apparently not supported by the windows compiler.  Switching to malloc as a short term solution to get the build working again, and will eventually use a mempool.

Fixed some small naming issues

Signed-off-by: Seth Zegelstein <szegel@amazon.com>

### CODE REVIEW

**Comment by darrylabbate** [comment]
Is there a measurable performance degradation with `malloc()`...`free()` over VLAs? If so, it may be worth implementing the `malloc()` only for Windows compilers

**Comment by a-szegel** [comment]
forgot free... whoops.

File: prov/efa/src/rxr/rxr_pkt_type_req.c [hunk 1]
```diff
@@ -2312,26 +2312,28 @@ void rxr_pkt_handle_write_rta_send_completion(struct rxr_ep *ep, struct rxr_pkt_
...
 	uint64_t device = efa_mr->peer.device.reserved;
 	int err;
 
 	/* Step 1: Copy data from device to temporary host buffer */
-	err = ofi_copy_from_hmem(efa_mr->peer.iface, device, host_data, iov->iov_base, iov->iov_len);
+	err = ofi_copy_from_hmem(efa_mr->peer.iface, device, host_data, dst->iov_base, dst->iov_len);
 	if (OFI_UNLIKELY(err)) {
+		free(host_data);
 		return err;
 	}
```
Comment (darrylabbate) [inline prov/efa/src/rxr/rxr_pkt_type_req.c:2327]: (1/2) Might be easier to maintain:

```suggestion
	if (OFI_UNLIKELY(err)) {
		goto done;
	}
```

File: prov/efa/src/rxr/rxr_pkt_type_req.c [hunk 2]
```diff
@@ -2312,26 +2312,28 @@ void rxr_pkt_handle_write_rta_send_completion(struct rxr_ep *ep, struct rxr_pkt_
...
 	ofi_atomic_write_handlers[op][dt](host_data,
 	                                  data,
-	                                  iov->iov_len / dtsize);
+	                                  dst->iov_len / dtsize);
 
 	/* Step 3: Copy temporary host buffer to device */
-	err = ofi_copy_to_hmem(efa_mr->peer.iface, device, iov->iov_base, host_data, iov->iov_len);
+	err = ofi_copy_to_hmem(efa_mr->peer.iface, device, dst->iov_base, host_data, dst->iov_len);
+	free(host_data);
 	return err;
```
Comment (darrylabbate) [inline prov/efa/src/rxr/rxr_pkt_type_req.c:2337]: (2/2)

```suggestion
done:
	free(host_data);
	return err;
```

**Comment by jtamzn** [comment]
use `alloca()` to allocate object on stack if VLA is not available.

**Comment by wzamazon** [comment]
This is try to make the CI stable. We can revisit whether there are way to improve performance later.

File: prov/efa/src/rxr/rxr_pkt_type_req.c (hunk 1 above)
Comment (wzamazon) [reply prov/efa/src/rxr/rxr_pkt_type_req.c:2327]: I agree with this. Though I will use `exit` instead of `done`. which

**Review by wzamazon — APPROVED** [review]

**Comment by wzamazon** [comment]
@darrylabbate  @jtamzn  I will merge this PR to make CI stable again. We will optimize later

**Pull request merged**